client-next/src/components/MapView/MapView.tsx

```diff
@@ -29,20 +33,43 @@ export function MapView({
   loading: boolean;
 }) {
   const onMapDoubleClick = useMapDoubleClick({ tripQueryVariables, setTripQueryVariables });
-  const [showPopup, setShowPopup] = useState<LngLat | null>(null);
+  const [showContextPopup, setShowContextPopup] = useState<LngLat | null>(null);
+  const [showPropsPopup, setShowPropsPopup] = useState<PopupData | null>(null);
+  const showFeaturePropPopup = (
+    e: mapboxgl.MapMouseEvent & {
+      features?: mapboxgl.MapboxGeoJSONFeature[] | undefined;
+    },
+  ) => {
+    if (e.features) {
+      // if you click on a cluster of map features it's possible that there are multiple
+      // to select from. we are using the first one instead of presenting a selection UI.
+      // you can always zoom in closer if you want to make a more specific click.
+      const feature = e.features[0];
+      setShowPropsPopup({ coordinates: e.lngLat, feature: feature });
+    }
+  };
 
   return (
     <div className="map-container below-content">
       <Map
         // @ts-ignore
         mapLib={import('maplibre-gl')}
         // @ts-ignore
-        mapStyle={mapStyle}
+        mapStyle={styleUrl}
         initialViewState={initialViewState}
         onDblClick={onMapDoubleClick}
         onContextMenu={(e) => {
-          setShowPopup(e.lngLat);
+          setShowContextPopup(e.lngLat);
+        }}
+        interactiveLayerIds={['regular-stop']}
+        onClick={(e) => {
```

Comment:
Now this is really getting nitpicky, but you could simplify this by removing the anonymous function and just pass `showFeaturePropPopup` directly as the value of the onClick prop.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5602/commits/cf5d875fbf4dd003023e4a506ac8a84d0175d8a7